### PR TITLE
Split lx planning inductor ops

### PR DESCRIPTION
**Pull request by spectre-ns**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [X] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Currently a truncated test set runs as a smoke test for issues related to LX planning over the current inductor ops tests. This was well reasoned at the time due to CI time constraints. Now we have the ability to plan out test shards using configuation YAML files. This allows hugely parallel execution of the full test sets to guard against regressions in tests outside the truncated set. 

Previously skipped tests have been moved into XFAIL modes in the configuration files. The skip machinery has been retained should a case arise that causes global side effects that corrupts other test cases. 

#### Which issue(s) this PR is related to:
https://github.com/torch-spyre/torch-spyre/issues/2062

#### Special notes for your reviewer:
Please note if there is a better pattern to accomplish this objective.

#### Does this PR introduce a user-facing change?
No. Testing only.

#### Additional note:
None.

### CODE REVIEW

**Comment by dgrove-oss** [comment]
@ashokponkumar @anubhavjana  -- does this look reasonable to you?

The desire is to run the full lx_planning_suite in CI to catch regressions, but to spread it across enough jobs that it doesn't add too much wall clock delay.

File: tests/inductor/test_inductor_ops_lx_planning.py [hunk 1]
```diff
@@ -32,14 +32,14 @@
...
-    os.environ.get("TEST_LX_PLANNING_RUN_SKIPS", "0") == "1"
+    os.environ.get("TEST_LX_PLANNING_RUN_SKIPS", "1") == "1"
 )
 
 # By default, only run one representative test per (prefix, op) cell of
 # TestOps.PARAMS plus all non-parameterized methods. Set this to "1" to
 # wrap every generated test — useful for thorough triage, skip-list
 # maintenance, and CI but slow for everyday dev workflow.
-tests_lx_planning_full: bool = os.environ.get("TEST_LX_PLANNING_FULL", "0") == "1"
+tests_lx_planning_full: bool = os.environ.get("TEST_LX_PLANNING_FULL", "1") == "1"
```
Comment (jturney) [inline tests/inductor/test_inductor_ops_lx_planning.py:42]: Do we want this to be enabled by default? Users running tests locally will still want to run the reduced set I suspect. Now we've just made it so that they have to set a flag to not do that.

File: tests/inductor/test_inductor_ops_lx_planning.py [hunk 2]
```diff
@@ -147,73 +147,7 @@ def new_test(self, value=value):
...
-    "test_unbind_2d_dim1",
-    "test_unbind_2d_dimneg1",
-    "test_unbind_3d_dim0",
-    "test_unbind_3d_dim1",
-    "test_unbind_3d_dim2",
-    "test_unbind_3d_dimneg1",
-    "test_unbind_4d_dim0",
-    "test_unbind_4d_dim3",
-]
+POINTWISE_TEST_FAILURES = []
```
Comment (jturney) [inline tests/inductor/test_inductor_ops_lx_planning.py:150]: Do we want to remove the ability to run the tests cleanly on our local boxes?

File: tests/inductor/test_inductor_ops_lx_planning.py (hunk 1 above)
Comment (dgrove-oss) [reply tests/inductor/test_inductor_ops_lx_planning.py:42]: Fair point.   Yes, we probably want to do something.  

To run locally, we do `make tests` with various configurations (https://github.com/torch-spyre/torch-spyre/blob/5ff24695fe0c917db658bd22558f517a73e91b78/Makefile#L12-L20).  

Since lx_planning is in `full` but not `core` this is probably ok, but maybe we need another config to toggle?  I don't have a strong opinion.

Comment (jturney) [reply tests/inductor/test_inductor_ops_lx_planning.py:42]: Ah, I've actually never used `make tests` locally. I just run pytest directly with the test file I'm working with and then let CI run everything else. I should look into this more.

File: tests/inductor/test_inductor_ops_lx_planning.py (hunk 2 above)
Comment (spectre-ns) [reply tests/inductor/test_inductor_ops_lx_planning.py:150]: I didn't think it was reasonable to maintain two lists. I added the xfail list to the configs.

@dgrove-oss can we execute locally with the OOT framework with the xfails respected?

File: tests/inductor/test_inductor_ops_lx_planning.py (hunk 1 above)
Comment (spectre-ns) [reply tests/inductor/test_inductor_ops_lx_planning.py:42]: @jturney  I did this so wouldn't need to inject the environment variable from test_matrix.py. Either default works for me.

File: tests/inductor/test_inductor_ops_lx_planning.py (hunk 2 above)
Comment (dgrove-oss) [reply tests/inductor/test_inductor_ops_lx_planning.py:150]: I believe that is what `make tests` does.

Comment (spectre-ns) [reply tests/inductor/test_inductor_ops_lx_planning.py:150]: @jturney  does this address your concern?

Comment (jturney) [reply tests/inductor/test_inductor_ops_lx_planning.py:150]: Yes, I believe it does 😀

File: tests/configs/torch_spyre_tests/inductor/test_inductor_ops_lx_matmul_config.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#  LxPlanningTwoOpReductionTest 'torch.sum(x, dim=0)').
+#
+# Auto-split of test_inductor_ops_lx_planning.py, mirroring the
+# test_inductor_ops.py config sharding. The full (non-truncated) test
+# set is enabled in CI via TEST_LX_PLANNING_FULL=1 + TEST_LX_PLANNING_RUN_SKIPS=1
+# (see .github/workflows/_test_matrix.yaml). The xfail lists below are
+# the source of truth for known LX-planning failures (the in-file .py
+# skip-lists were removed as duplicates).
+test_suite_config:
+  labels: [core, full]
```
Comment (dgrove-oss) [inline tests/configs/torch_spyre_tests/inductor/test_inductor_ops_lx_matmul_config.yaml]: I think we want this to be just `[full]` not `[core, full]` across all the shards to mirror the old behavior for people running tests locally.

**Review by dgrove-oss — DISMISSED** [review]

File: tests/configs/torch_spyre_tests/inductor/test_inductor_ops_lx_matmul_config.yaml (hunk 3 above)
Comment (spectre-ns) [reply tests/configs/torch_spyre_tests/inductor/test_inductor_ops_lx_matmul_config.yaml]: Makes sense. The requested change has been applied across all files.

**Review by dgrove-oss — APPROVED** [review]
lgtm
